File: src/handlers/claimHandler.js
import { client } from "../bot.js";
import { ENV } from "../config/env.js";
import { BUTTONS } from "../config/constants.js";
import { EmbedBuilder, PermissionFlagsBits } from "discord.js";
import { supabase } from "../database/supabase.js";

// Handles Claim & Decline button interactions
export const initClaimHandler = () => {
  client.on("interactionCreate", async (interaction) => {
    if (!interaction.isButton()) return;

    const { customId, channel, user } = interaction;

    // ===== CLAIM BUTTON =====
    if (customId === BUTTONS.CLAIM) {
      try {
        // Delete the original order embed
        await interaction.message.delete();

        // Create private order-ticket channel
        const ticketChannel = await channel.guild.channels.create({
          name: `order-${user.username}`,
          type: 0, // GUILD_TEXT
          parent: ENV.ORDER_CATEGORY_ID,
          permissionOverwrites: [
            {
              id: user.id, // claiming chef
              allow: [PermissionFlagsBits.ViewChannel, PermissionFlagsBits.SendMessages]
            },
            {
              id: interaction.guild.id, // everyone else
              deny: [PermissionFlagsBits.ViewChannel]
            }
          ]
        });

        // Post order details
        const embed = new EmbedBuilder()
          .setTitle("Order Claimed")
          .setDescription(`Chef: ${user}\nOrder ID: TBD`)
          .setColor("Green")
          .setTimestamp();

        await ticketChannel.send({ content: `<@customer_id_here>`, embeds: [embed] });

        // TODO: Insert order into Supabase with status 'claimed'

        await interaction.reply({ content: "Order claimed and private channel created!", ephemeral: true });
      } catch (err) {
        console.error("Claim handler error:", err);
      }
    }

    // ===== DECLINE BUTTON =====
    if (customId === BUTTONS.DECLINE) {
      try {
        await interaction.message.delete();
        await interaction.reply({ content: "Order declined.", ephemeral: true });
      } catch (err) {
        console.error("Decline handler error:", err);
      }
    }
  });
};

File: src/handlers/claim_handler.py
import nextcord
from nextcord import Interaction, ChannelType, PermissionOverwrite
from nextcord.ui import View
from src.config.env import ORDER_CATEGORY_ID
from src.utils.permissions import create_ticket_permissions
from src.utils.order_id import generate_order_id

async def claim_callback(interaction: Interaction, order_content: str, customer_id: int):
    guild = interaction.guild
    chef = interaction.user

    # Delete original embed message
    await interaction.message.delete()

    # Create private ticket channel
    overwrites = {
        guild.default_role: PermissionOverwrite(view_channel=False),
        chef: PermissionOverwrite(view_channel=True, send_messages=True),
        guild.get_member(customer_id): PermissionOverwrite(view_channel=True, send_messages=True)
    }

    category = guild.get_channel(ORDER_CATEGORY_ID)
    ticket_channel = await guild.create_text_channel(
        name=f"order-{chef.name}",
        category=category,
        overwrites=overwrites,
        type=ChannelType.text
    )

    # Post order details
    await ticket_channel.send(
        f"📌 **Order ID:** {generate_order_id()}\n"
        f"👨‍🍳 **Claimed by:** {chef.mention}\n"
        f"📝 **Order Details:** {order_content}\n"
        f"📢 <@{customer_id}>, your order has been claimed!"
    )

    # Respond to chef
    await interaction.response.send_message(f"✅ Order claimed! Check {ticket_channel.mention}", ephemeral=True)
